MaxText/layers/moe.py

```diff
@@ -914,7 +928,7 @@ def __call__(self, inputs):
         kernel_axes=self.kernel_axes,
         name="gate",
         use_bias=cfg.routed_bias,
-        score_func=cfg.routed_score_func,
+        score_func=cfg.routed_score_func if not is_llama4_decoder_block else None,
```

Comment:
We could skip the `routed_score_func` as default (no need to add `if`). Remove `routed_score_func` from config.